### ISSUE TITLE
Connection closes when calling child process

### ISSUE THREAD

**Issue by Nolence** [websockets/ws#1457]
#### Description

This problem took me forever to find in my code and I haven't been able to find a solution so apologies if it's dumb.

After creating a WebSocket connection, I have the client send a message (a plain url) to the server where the onmessage function takes over. It calls a function called download:

```javascript
const download = (url, destination) => {
  let child = spawn("curl", [url, "--output", destination]);
  child.on("error", err => {});
  child.on("close", (code, signal) => {});
};
```

The download function closes the connection with a code 1001. Why is this closing the connection? What can I do so it doesn't close?

Any help is appreciated

#### Reproducible in:

version: 6.1.0
Node.js version(s): 10.4.1
OS version(s): Windows 10 (Git Bash Shell)

**Comment by lpinca** [comment]
If I understand correctly the `'message'` event listener on the server calls `download()` which spawns a new process right? I doubt this the reason why the connection is closed, it doesn't make much sense.

You get a 1001 status codes which is usually sent by browsers when the page is refreshed.

**Comment by Nolence** [comment]
Right, the `'message'` event listener on the server calls `download()` and from there, I immediately get an error from the `onerror` listener. I have been looking into what might be the issue to narrow it down even further and I found that the same thing even happens if I use another library such as Socket.IO.

If I call simpler child process like `ls` the connection is not closed. It's only when I try downloading (I have tried three different methods and they all break the connection). My understanding was that spawning child processes didn't block the event loop, but that's what it looks like is happening???

The weird 1001 status page refreshing code is even more confusing to me

**Comment by Nolence** [comment]
Sheesh, that was a fun trip down the rabbit hole. I Found the problem. I was running the live server extension from Visual Studio Code and I guess every time I was spawning a child process, it was restarting live-server... hence the 1001 refresh code.

It probably would've taken me way way longer to find this bug if you hadn't mentioned the refresh thing so thank you!!
